swmm-toolkit/src/swmm/toolkit/solver.i

```diff
@@ -30,23 +39,277 @@
 /* GENERATES DOCUMENTATION */
 %feature("autodoc", "2");
 
-
 /* RENAME FUNCTIONS PYTHON STYLE */
 %include "solver_rename.i"
 
-
 /* INSERTS CUSTOM EXCEPTION HANDLING IN WRAPPER */
 %exception
 {
     $function
     if (result > 0) {
-        char errmsg[90];
-        swmm_getError(errmsg, 90);
-        PyErr_SetString(PyExc_Exception, errmsg);
+        char* errorMsg = NULL;
```

Comment:
The SWIG wrapper should throw an exception with the SWMM error message where there is an error. Have you tested?

Reply:
Yes, there is an unit test to test for an Exception. I will add a check for the exception message.

Reply:
@cbuahin, at the moment there are no api error messages for warning to my knowledge.